src/stdlib_str2num.fypp

```diff
@@ -0,0 +1,569 @@
+!>   difference abs :                                 0.4625E-475 
+!>   difference rel :                                 0.3300E-029%
+
+module stdlib_str2num
+    use stdlib_kinds, only: sp, dp, xdp, qp, int8, int16, int32, int64
+    use, intrinsic:: ieee_arithmetic, only: ieee_value, ieee_positive_inf, ieee_quiet_nan
+    implicit none
+    private
+    public :: to_num, to_num_p
+    
+    integer(int8), parameter :: digit_0    = ichar('0',int8)
+    integer(int8), parameter :: period     = ichar('.',int8) - digit_0
+    integer(int8), parameter :: comma      = ichar(',',int8) - digit_0
+    integer(int8), parameter :: minus_sign = ichar('-',int8) - digit_0
+    integer(int8), parameter :: plus_sign  = ichar('+',int8) - digit_0
+    integer(int8), parameter :: Inf        = ichar('I',int8) 
+    integer(int8), parameter :: NaN        = ichar('N',int8) 
+    integer(int8), parameter :: le         = ichar('e',int8) - digit_0
+    integer(int8), parameter :: BE         = ichar('E',int8) - digit_0
+    integer(int8), parameter :: ld         = ichar('d',int8) - digit_0
+    integer(int8), parameter :: BD         = ichar('D',int8) - digit_0
+    integer(int8), parameter :: LF = 10, CR = 13, WS = 32
+
+    interface to_num
+        !! version: experimental
+        !!
+        !! Conversion of strings to numbers
+        !! ([Specification](../page/specs/stdlib_str2num.html#to-num-conversion-of-strings-to-numbers)) 
+        #:for k1, t1 in (INT_KINDS_TYPES + REAL_KINDS_TYPES)
+        module procedure to_${k1}$
+        #:endfor
+    end interface
+
+    interface to_num_p
+        !! version: experimental
+        !!
+        !! Conversion of a stream of values in a string to numbers
+        !! ([Specification](../page/specs/stdlib_str2num.html#to-num-p-conversion-of-a-stream-of-values-in-a-strings-to-numbers)) 
+        #:for k1, t1 in (INT_KINDS_TYPES + REAL_KINDS_TYPES)
+        module procedure to_${k1}$_p
+        #:endfor
+    end interface
+
+    interface to_num_base
+        #:for k1, t1 in (INT_KINDS_TYPES + REAL_KINDS_TYPES)
+        module procedure to_${k1}$_base
+        #:endfor
+    end interface
+    
+    contains
+    
+    !---------------------------------------------
+    ! String To Number interfaces
+    !---------------------------------------------
+
+    #:for k1, t1 in (INT_KINDS_TYPES + REAL_KINDS_TYPES)
+    elemental function to_${k1}$(s,mold) result(v)
+        ! -- In/out Variables
+        character(*), intent(in) :: s !> input string
+        ${t1}$, intent(in) :: mold !> dummy argument to disambiguate at compile time the generic interface
+        ${t1}$ :: v !> Output ${t1}$ value
+        ! -- Internal Variables
+        integer(int8) :: p !> position within the number
+        integer(int8)  :: stat !> error status
+        !----------------------------------------------
+        call to_num_base(s,v,p,stat)
+    end function
+    
+    function to_${k1}$_p(s,mold,stat) result(v)
```

Comment:
Does `_p` here refer to pointer? If yes, `_ptr` or `_pointer` will be clearer for the user.

But "pointer" is also something implementation specific, rather than describing what the function does.

Should it be `to_num_from_stream` or similar?

Reply:
Yes, the `_p` if related to the function taking as input a pointer to a character. This `_ptr` looks good,  this also `to_num_from_stream`, why not. I like to find the balance between explicit names and not-too-long but this is fine for me. If no objection from others, I'll change then for your last suggestion @milancurcic.

Reply:
done